### PR TITLE
Add test coverage for slicing with "out of bounds" negative indices

### PR DESCRIPTION
**Pull request by wence-**
## Description

Polars wraps negative starts and then clamps both the resulting start
and length to [0, num_rows), so we should do that.

Add tests of this behaviour as well.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by wence-** [comment]
Before:
```
---------- coverage: platform linux, python 3.10.14-final-0 ----------
Name                                  Stmts   Miss  Cover
---------------------------------------------------------
TOTAL                                  1322    249    81%
```
After:
```
--------- coverage: platform linux, python 3.10.14-final-0 ----------
Name                                  Stmts   Miss  Cover
---------------------------------------------------------
TOTAL                                  1328    247    81%
```

File: python/cudf_polars/tests/test_slice.py [hunk 1]
```diff
@@ -11,13 +11,14 @@
...
+    [0, 1, 2, -10, -20, -1, -2, 20],
 )
 @pytest.mark.parametrize(
     "len",
-    [0, 2, 12],
+    [0, 2, 12, 11],
 )
-def test_slice(offset, len):
+@pytest.mark.parametrize("slice_pushdown", [False, True])
+def test_slice(offset, len, slice_pushdown):
```
Comment (bdice) [inline python/cudf_polars/tests/test_slice.py:21]: Avoid shadowing the Python built-in `len`. I would rename this parameter to `length` to match the Polars API: https://docs.pola.rs/api/python/stable/reference/dataframe/api/polars.DataFrame.slice.html

**Review by bdice — APPROVED** [review]
One naming request, otherwise LGTM.

**Pull request merged**